changes.go

```diff
@@ -139,3 +158,29 @@ func handleRelations(add addChangeFunc, relations [][]string, addedServices map[
 func handleUnits(add addChangeFunc, services map[string]*charm.ServiceSpec, addedServices, addedMachines map[string]string) {
 	// TODO frankban: implement this.
 }
+
+// parseEndpoint creates an endpoint from its string representation.
+func parseEndpoint(e string) *endpoint {
```

Comment:
For now this is only intended as a helper for strings manipulation. But I agree with you that having shared code for endpoints would be useful.